lib/logstash/inputs/mongodb.rb

```diff
@@ -144,10 +144,15 @@ def get_collection_names(mongodb, collection)
   end
 
   public
-  def get_cursor_for_collection(mongodb, mongo_collection_name, last_id_object, batch_size)
+  def get_cursor_for_collection(mongodb, mongo_collection_name, first_entry_id_object, last_id_object, batch_size)
```

Comment:
We should probably rename these variables as they're a bit confusing.

Maybe...
first_entry_id_object => processing_start_entry_id_object
last_id_object => last_unprocessed_entry_id_object (or maybe first_unprocessed_entry_id_object)

... or something like that.